src/yunohost/certificate.py

```diff
@@ -0,0 +1,641 @@
+    """
+    Renew Let's Encrypt certificate for given domains (all by default)
+
+    Keyword argument:
+        domainList -- Domains for which to renew the certificates
+        force      -- Ignore the validity threshold (15 days)
+        no-check   -- Disable some checks about the reachability of web server
+                      before attempting the renewing
+        email      -- Emails root if some renewing failed
+    """
+
+    # If no domains given, consider all yunohost domains with Let's Encrypt
+    # certificates
+    if (domainList == []) :
+        for domain in yunohost.domain.domain_list(auth)['domains'] :
+
+            # Does it has a Let's Encrypt cert ?
+            status = _get_status(domain)
+            if (status["CAtype"] != "Let's Encrypt") : continue
+
+            # Does it expires soon ?
+            if (force) or (status["validity"] <= validity_limit) :
+                domainList.append(domain)
+
+        if (len(domainList) == 0) :
+            logger.info("No certificate needs to be renewed.")
+
+    # Else, validate the domain list given
+    else :
+        for domain in domainList :
+
+            # Is it in Yunohost dmomain list ?
+            if domain not in yunohost.domain.domain_list(auth)['domains']:
+                raise MoulinetteError(errno.EINVAL, m18n.n('certmanager_domain_unknown', domain=domain))
+            
+            status = _get_status(domain)
+
+            # Does it expires soon ?
+            if not ((force) or (status["validity"] <= validity_limit)) :
+                raise MoulinetteError(errno.EINVAL, m18n.n('certmanager_attempt_to_renew_valid_cert', domain=domain))
+
+
+            # Does it has a Let's Encrypt cert ?
+            if (status["CAtype"] != "Let's Encrypt") :
+                raise MoulinetteError(errno.EINVAL, m18n.n('certmanager_attempt_to_renew_nonLE_cert', domain=domain))
+
+
+    # Actual renew steps
+    for domain in domainList :
+        
+        logger.info("Now attempting renewing of certificate for domain "+domain+" !")
+
+        try : 
+
+            if not no_checks : _check_domain_is_correctly_configured(domain)
+            _backup_current_cert(domain)
+            _fetch_and_enable_new_certificate(domain)
+
+            logger.success(m18n.n("certmanager_cert_renew_success", domain=domain))
+    
+        except Exception as e :
+           
+            logger.error("Certificate renewing for "+domain+" failed !")
+            logger.error(str(e))
+
+            if (email) :
+                logger.error("Sending email with details to root ...")
+                _email_renewing_failed(domain, e)
+            
+
+
+
+
+###############################################################################
+#   Back-end stuff                                                            #
+###############################################################################
+
+def _install_cron() :
+
+    cron_job_file = "/etc/cron.weekly/certificateRenewer"
```

Comment:
Maybe rename cron file to something more consistent with other yunohost cron. Currently we have `/etc/cron.d/yunohost-applist-yunohost`,  `/etc/cron.d/yunohost-firewall` and `/etc/cron.d/yunohost-dyndns`.
I don't know if it worth to move this cron to `/etc/cron.d` as well.

I vote for `/etc/cron.d/yunohost-certificate-renew` ;)